app/frontend/components/common/unexpectedErrorModal.tsx

```diff
@@ -1,84 +1,81 @@
-import {h, Component} from 'preact'
+import {h} from 'preact'
+import {useState, useCallback} from 'preact/hooks'
 import {connect} from '../../helpers/connect'
 import actions from '../../actions'
 import Modal from './modal'
 import Alert from './alert'
+import submitFeedbackToSentry from '../../helpers/submitFeedbackToSentry'
 
 interface Props {
   sendSentry: any
   closeUnexpectedErrorModal: () => void
-  updateName: any
-  updateEmail: any
-  updateMessage: any
-  submitUserFeedbackToSentry: () => void
 }
 
-class UnexpectedErrorModal extends Component<Props, {}> {
-  constructor(props) {
-    super(props)
-    this.closeAndResolve = this.closeAndResolve.bind(this)
-  }
+const UnexpectedErrorModal = ({sendSentry, closeUnexpectedErrorModal}) => {
+  const [userEmail, setEmail] = useState('')
+  const [userName, setName] = useState('')
+  const [userComments, setComments] = useState('')
 
-  closeAndResolve(resolveValue) {
-    this.props.sendSentry.resolve(resolveValue)
-    this.props.closeUnexpectedErrorModal()
-  }
+  const closeAndResolve = useCallback(
+    (shouldSend: boolean) => {
+      // first tell sentry to send the event
+      sendSentry.resolve(shouldSend) // TODO(ppershing): is this a promise or what?
```

Comment:
@PeterBenc , can you elaborate?

Reply:
yes, it resolves a promise, the promise is pending in `beforeSend` function in `Sentry.init`. Whether the event is send to setry or not is based on the value of the resolved promise.

Reply:
in walletApp.js
```
beforeSend(event) {
    return new Promise((resolve) => {
      store.setState({
        sendSentry: {
          event,
          resolve,
        },
        showUnexpectedErrorModal: true,
      })
    }).then((res) => {
      return res === true ? event : null
    })
  },